File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb
' Visual Basic .NET Document
Option Strict On

Module Example
    Public Sub Main()
        ' <Snippet22>
        Dim position As Integer
        Dim softHyphen As String = ChrW(&HAD)
        Dim s1 As String = "ani" + softHyphen + "mal"
        Dim s2 As String = "animal"

        ' Find the index of the soft hyphen followed by "n".
        position = s1.LastIndexOf("m")
        Console.WriteLine($"'m' at position {position}")

        If position >= 0 Then
            Console.WriteLine(s1.LastIndexOf(softHyphen + "n", position))
        End If

        position = s2.LastIndexOf("m")
        Console.WriteLine($"'m' at position {position}")

        If position >= 0 Then
            Console.WriteLine(s2.LastIndexOf(softHyphen + "n", position))
        End If

        ' Find the index of the soft hyphen followed by "m".
        position = s1.LastIndexOf("m")
        Console.WriteLine($"'m' at position {position}")

        If position >= 0 Then
            Console.WriteLine(s1.LastIndexOf(softHyphen + "m", position))
        End If

        position = s2.LastIndexOf("m")
        Console.WriteLine($"'m' at position {position}")

        If position >= 0 Then
            Console.WriteLine(s2.LastIndexOf(softHyphen + "m", position))
        End If

        ' The example displays the following output:
        '
        ' 'm' at position 4
        ' 1
        ' 'm' at position 3
        ' 1
        ' 'm' at position 4
        ' 4
        ' 'm' at position 3
        ' 3
        ' </Snippet22>
    End Sub
End Module

File: snippets/fsharp/System/String/LastIndexOf/lastindexof22.fs
module lastindexof22.fs
do
    // <Snippet22>
    let s1 = "ani\u00ADmal"
    let s2 = "animal"

    // Find the index of the soft hyphen followed by "n".
    let position = s1.LastIndexOf "m"
    printfn $"'m' at position {position}"

    if position >= 0 then
        printfn $"""{s1.LastIndexOf("\u00ADn", position)}"""

    let position = s2.LastIndexOf "m"
    printfn $"'m' at position {position}"

    if position >= 0 then
        printfn $"""{s2.LastIndexOf("\u00ADn", position)}"""

    // Find the index of the soft hyphen followed by "m".
    let position = s1.LastIndexOf "m"
    printfn $"'m' at position {position}"

    if position >= 0 then
        printfn $"""{s1.LastIndexOf("\u00ADm", position)}"""

    let position = s2.LastIndexOf "m"
    printfn $"'m' at position {position}"

    if position >= 0 then
        printfn $"""{s2.LastIndexOf("\u00ADm", position)}"""
    // The example displays the following output:
    //
    // 'm' at position 4
    // 1
    // 'm' at position 3
    // 1
    // 'm' at position 4
    // 4
    // 'm' at position 3
    // 3
    // </Snippet22>
